packages/analytics-js-plugins/src/errorReporting/event/event.ts

```diff
@@ -64,80 +66,33 @@ const hasNecessaryFields = (error: any) =>
   (typeof error.name === 'string' || typeof error.errorClass === 'string') &&
   (typeof error.message === 'string' || typeof error.errorMessage === 'string');
 
-const normaliseError = (
-  maybeError: any,
-  tolerateNonErrors: boolean,
-  component: string,
-  logger?: ILogger,
-) => {
+const normaliseError = (maybeError: any, component: string, logger?: ILogger) => {
   let error;
   let internalFrames = 0;
 
-  const createAndLogInputError = (reason: string) => {
-    const verb = component === 'error cause' ? 'was' : 'received';
-    if (logger) logger.warn(`${component} ${verb} a non-error: "${reason}"`);
-    const err = new Error(
-      `${component} ${verb} a non-error. See "${component}" tab for more detail.`,
-    );
-    err.name = 'InvalidError';
-    return err;
-  };
-
-  // In some cases:
-  //
-  //  - the promise rejection handler (both in the browser and node)
-  //  - the node uncaughtException handler
-  //
-  // We are really limited in what we can do to get a stacktrace. So we use the
-  // tolerateNonErrors option to ensure that the resulting error communicates as
-  // such.
-  if (!tolerateNonErrors) {
-    if (isError(maybeError)) {
-      error = maybeError;
-    } else {
-      error = createAndLogInputError(typeof maybeError);
-      internalFrames += 2;
-    }
+  if (typeof maybeError === 'object' && isError(maybeError)) {
```

Comment:
I did not suggest the object check here but for the next else if.

```suggestion
  if (isError(maybeError)) {
```

Reply:
Addressed